### PR TITLE
fix: segment overrides stale feature state value while creating GitHub comment

### PR DESCRIPTION
**Pull request by novakzaballa**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Move GitHub task out of lifecycle hook to serializer
- Used the `AFTER_SAVE` hook within the `feature `model to create a GitHub comment when the feature is deleted (we use `AFTER_SAVE` because it's a soft delete).

## How did you test this code?

- Add unit tests

### CODE REVIEW

File: api/integrations/github/github.py [hunk 1]
```diff
@@ -89,26 +90,29 @@ def generate_body_comment(
...
             feature_id,
             tab,
         )
-        if v.get("segment_name") is not None and v["segment_name"] != last_segment_name:
-            result += "\n" + LINK_SEGMENT_TITLE % (v["segment_name"])
-            last_segment_name = v["segment_name"]
+        if (
+            fs.get("segment_name") is not None
+            and fs["segment_name"] != last_segment_name
+        ):
```
Comment (zachaysan) [inline api/integrations/github/github.py:106]: Assuming that `last_segment_name` is never `None` this should simplify to the below.

```suggestion
        if (
            fs.get("segment_name") != last_segment_name
        ):
```

File: api/tests/unit/features/test_unit_feature_external_resources_views.py [hunk 2]
```diff
@@ -10,16 +10,52 @@
...
+    feature_state_enabled="❌ Disabled",
+    feature_state_value="`value`",
+) -> str:
+    return (
+        "| Environment | Enabled | Value | Last Updated (UTC) |\n"
+        "| :--- | :----- | :------ | :------ |\n"
+        f"| [Test Environment](https://example.com/project/{project_id}/"
+        f"environment/{environment_api_key}/features?feature={feature_id}&tab=value) "
+        f"| {feature_state_enabled} | {feature_state_value} | {feature_state_updated_at} |\n"
+    )
```
Comment (zachaysan) [inline api/tests/unit/features/test_unit_feature_external_resources_views.py:38]: Can you double check this? I don't think the data will line up in columns properly but maybe you already know this.

File: api/tests/unit/features/test_unit_feature_external_resources_views.py [hunk 3]
```diff
@@ -10,16 +10,52 @@
...
+
+
+def expected_segment_comment_body(
+    project_id,
+    environment_api_key,
+    feature_id,
+    segment_override_updated_at,
+    segment_override_enabled,
+    segment_override_value,
+) -> str:
```
Comment (zachaysan) [inline api/tests/unit/features/test_unit_feature_external_resources_views.py]: Missing typing

File: api/tests/unit/features/test_unit_feature_external_resources_views.py [hunk 4]
```diff
@@ -10,16 +10,52 @@
...
 _django_json_encoder_default = DjangoJSONEncoder().default
 
 
+def expected_default_body(
+    project_id,
+    environment_api_key,
+    feature_id,
+    feature_state_updated_at,
+    feature_state_enabled="❌ Disabled",
+    feature_state_value="`value`",
```
Comment (zachaysan) [inline api/tests/unit/features/test_unit_feature_external_resources_views.py]: Missing typing

**Review by zachaysan — APPROVED** [review]
Some comments but nothing show-stopping. Make sure to add the typing though.

File: api/integrations/github/github.py (hunk 1 above)
Comment (novakzaballa) [reply api/integrations/github/github.py:106]: fs.get("segment_name") can be None

File: api/tests/unit/features/test_unit_feature_external_resources_views.py (hunk 4 above)
Comment (novakzaballa) [reply api/tests/unit/features/test_unit_feature_external_resources_views.py]: Done

File: api/tests/unit/features/test_unit_feature_external_resources_views.py (hunk 3 above)
Comment (novakzaballa) [reply api/tests/unit/features/test_unit_feature_external_resources_views.py]: Done

File: api/tests/unit/features/test_unit_feature_external_resources_views.py (hunk 2 above)
Comment (novakzaballa) [reply api/tests/unit/features/test_unit_feature_external_resources_views.py:38]: Double-check is ok, I attached an image:
image

**Pull request merged**